src/codegen/block.rs

```diff
@@ -235,6 +249,9 @@ pub fn compile_block<'a, 'b>(
                 b.build_br(tt.bb);
                 // Fetch the ctx
                 let if_ctx_resolved = if_ctx.as_ref().expect("malformed wasm -- else with no if");
+                // Ensure we don't build a duplicate else
+                assert_eq!(if_ctx_resolved.else_built.get(), false);
```

Comment:
Can we spit out a useful error message here even if we do just end up bombing out?